src/shared/ranking.js

```diff
@@ -0,0 +1,33 @@
+/**
+ * This function get data from localStorage by category name.
+ * @param {string} category
+ * @return {object} {"category","userData":[{"username","points"}]}
```

Comment:
```suggestion
 * @return {{ category: string, userData: { username: string, points: number }[] }}
```

Ups, źle zrozumiałem co chcesz stąd zwracać na podstawie poprzedniego komentarza. W takim razie będzie coś w stylu `@return {{ category: string, userData: { username: string, points: number }[] }}`.

Rozpiszę, żebyś mniej więcej wiedział co się dzieje:
- zewnętrzne nawiasy klamrowe są z JSDoca
- wewnętrzne nawiasy oznaczają obiekt, który ma właściwości:
  - `category` typu `string`
  - `userData` typu tablica obiektów, które mają właściwości:
    - `username` typu `string`
    - `points` typu `number`

To co zmieniłeś w poprzednim commicie oznaczało, że zwracany jest obiekt, ale wszystkie informacje na temat jego "struktury" były poza zewnętrznymi `{` i `}`, więc były traktowane jako informacje dodatkowe i ignorowane przez system typizacji.